src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -516,6 +529,10 @@ private async Task ExecuteInternal()
                 {
                     // Subscribe to database informational messages
                     sqlConn.GetUnderlyingConnection().InfoMessage -= OnInfoMessage;
+                    var intPID = (sqlConn.GetUnderlyingConnection() as SqlConnection).ServerProcessId;
```

Comment:
Why are you duplicating the logic here if we're already doing it up above? It seems like this one should be removed - since I believe we need to set the PID before the QueryCompleted callback is done